src/lib.rs

```diff
@@ -609,221 +407,108 @@ impl Renderer {
             StencilFunc: D3D11_COMPARISON_ALWAYS,
         };
         let desc = D3D11_DEPTH_STENCIL_DESC {
-            DepthEnable: FALSE,
+            DepthEnable: false.into(),
             DepthWriteMask: D3D11_DEPTH_WRITE_MASK_ALL,
             DepthFunc: D3D11_COMPARISON_ALWAYS,
-            StencilEnable: FALSE,
+            StencilEnable: false.into(),
             StencilReadMask: 0,
             StencilWriteMask: 0,
             FrontFace: stencil_op_desc,
             BackFace: stencil_op_desc,
         };
-        let depth_stencil_state = com_ptr_from_fn(|depth_stencil_state| {
-            device.CreateDepthStencilState(&desc, depth_stencil_state)
-        })?;
+        let depth_stencil_state = device.CreateDepthStencilState(&desc)?;
         Ok((blend_state, rasterizer_state, depth_stencil_state))
     }
 }
 
 #[derive(Debug)]
-struct Buffer(ComPtr<ID3D11Buffer>, usize);
+struct Buffer(pub Option<ID3D11Buffer>, usize);
```

Comment:
This shouldn't be nullable, if we have none here we failed initializating something which should return an error

Reply:
Done and done, I removed the pub as well